numpy/core/_methods.py

```diff
@@ -187,16 +188,49 @@ def _var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
     else:
         arrmean = arrmean.dtype.type(arrmean / rcount)
 
-    # Compute sum of squared deviations from mean
-    # Note that x may not be inexact and that we need it to be an array,
-    # not a scalar.
-    x = asanyarray(arr - arrmean)
-    if issubclass(arr.dtype.type, (nt.floating, nt.integer)):
-        x = um.multiply(x, x, out=x)
-    else:
-        x = um.multiply(x, um.conjugate(x), out=x).real
+    if (rcount > 1 and arr.ndim and
+            issubclass(arr.dtype.type, (nt.floating, nt.complexfloating))):
+        # Since sumsquarediff only works on the last axis, we need to transpose
+        if axis is None:
+            axis = tuple(range(arr.ndim))
+        elif isinstance(axis, int):
+            axis = (axis, )
+        else:
+            axis = tuple(sorted(axis))
+
+        S1, S2 = umt.sumsquarediff(
+            arr.swapaxes(axis[-1], -1),
+            arrmean.swapaxes(axis[-1], -1),
+            keepdims=keepdims,
+            dtype=dtype
+        )
 
-    ret = umr_sum(x, axis, dtype, out, keepdims)
+        # sum over the remaining axes
+        if len(axis) > 1:
+            S1 = umr_sum(S1, axis=axis[:-1], keepdims=keepdims)
+            S2 = umr_sum(S2, axis=axis[:-1], out=out, keepdims=keepdims)
+
+        if keepdims:
+            S1 = S1.swapaxes(axis[-1], -1)
+            S2 = S2.swapaxes(axis[-1], -1)
+
```

Comment:
This is still invalid... If swap the axes instead of rolling, when keepdims is False it doesn't work. I still haven't figure out how to call broadcast_to from numpy.core.